src/riscv/lib/src/machine_state/page_cache/router.rs

```diff
@@ -303,13 +306,17 @@ mod tests {
         }
     }
 
-    impl<T: Clone + RouterEq> Router<T> {
+    impl<T: Clone + RouterEq + Default> Router<T> {
         fn ranges(&self) -> Vec<RangeInclusive<u64>> {
             self.internal
                 .iter()
                 .map(|(r, _)| r.clone())
                 .collect::<Vec<_>>()
         }
+
+        fn add_range_default(&mut self, range: RangeInclusive<u64>) {
+            self.add_range(range, T::default);
+        }
```

Comment:
```suggestion
    impl<T: Clone + RouterEq> Router<T> {
        fn ranges(&self) -> Vec<RangeInclusive<u64>> {
            self.internal
                .iter()
                .map(|(r, _)| r.clone())
                .collect::<Vec<_>>()
        }

        fn add_range_default(&mut self, range: RangeInclusive<u64>)
        where
            T: Default {
            self.add_range(range, T::default);
        }
```
mega nit (I know it's test code) - but constraints can be more precise